gtest_parallel.py

```diff
@@ -302,7 +303,14 @@ class FilterFormat(object):
   def __init__(self, output_dir):
     if sys.stdout.isatty():
       # stdout needs to be unbuffered since the output is interactive.
-      sys.stdout = os.fdopen(sys.stdout.fileno(), 'w', 0)
+      if isinstance(sys.stdout, io.TextIOWrapper):
```

Comment:
I'd like to get rid of this block in virtually any way possible. Is there no way to get the bytes off of here?

Reply:
@pbos i am afraid that this fix is the best way i have in my mind to have unbuffered stdout in python3 =(

Reply:
@pbos is there anything i can do to progress this PR?

Reply:
Sorry, I was swamped since last time and wanted to have a chance to repro the errors you got when running. I'm able to run `python3 gtest-parallel /path/to/foo` with your changes and were not able to before. It seems like previous PRs to fix Py3 compatibility either had vastly different environments from ours or never really tested their fixes (I had no idea this was completely broken).

Long story short, thanks! Merging.